src/codegate/pipeline/base.py

```diff
@@ -260,19 +261,25 @@ def get_last_user_message_block(
         messages = request["messages"]
         block_start_index = None
 
+        base_tool = get_tool_name_from_messages(request)
+        accepted_roles = ["user", "assistant"]
+        if base_tool == "open interpreter":
+            # open interpreter also uses the role "tool"
+            accepted_roles.append("tool")
```

Comment:
sorry to be difficult about this review.

Would it break anything if we accepted the tools either way, meaning for all clients?

What do we miss if we don't process the tool messages here?

Reply:
well i was not aware if any other client using tools but wanted to be conservative about it, so that's why i filter by the client. Do you prefer that i open that and we do not filter?